File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        '''
        create a set from 1 to n where n = len(nums).
        check each num in nums if the num exists in sets then remove the num from the sets.
        '''
        # 1. create a set from 1 to n
        sets = set()
        n = len(nums) + 1
        for i in range(1, n):
            sets.add(i)
            
        # 2. check does num in nums exists in set
        for num in nums:
            if num in sets:
                sets.remove(num)
        
        return list(sets)

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        unordered_set<int> sets;

        for (int i = 1; i <= nums.size(); i++) {
            sets.insert(i);
        }

        for (int num : nums) {
            if (sets.find(num) != sets.end()) {
                sets.erase(num);
            }
        }

        vector<int> res(sets.begin(), sets.end());
        return res;
    }
};